### PR TITLE
fix(tooltip): tooltip margin bug

### LINKED ISSUE #8275: Space between icon and tooltip in Carbon tooltip component does not match sketch kit
## Space between icon and tooltip in Carbon tooltip component does not match sketch kit 

- Should have 8px between icon and tooltip but looks like has 0 or 1px

## What package(s) are you using?

- [X] `carbon-components-react`

### Actual in Carbon React

image (2)

### Expected in Carbon Sketch it

Screen Shot 2021-04-01 at 12 44 02 PM

### PR DESCRIPTION
**Pull request by andreancardona**
Closes #8275 

Adds small 8px margin to tooltip to match current design in Sketch kit

### CODE REVIEW

**Review by laurenmrice — CHANGES_REQUESTED** [review]
The alignments are off when you change the `direction` prop to `left` , `right` and `top`. After fixing those I would also check them against the `alignment` prop options as well.

Everything else looks good though!

tooltip

File: packages/components/src/components/tooltip/_tooltip.scss [hunk 1]
```diff
@@ -545,6 +545,7 @@
 
   .#{$prefix}--tooltip--shown {
     display: block;
+    margin-top: $spacing-03;
```
Comment (tw15egan) [inline packages/components/src/components/tooltip/_tooltip.scss]: ~I think we'll need to add these changes to `src/globals/scss/_tooltip.scss` to account for all the alignment variants~

Looks like we can fix this by adding the correct margin to the applicable `&[data-floating-menu-direction='DIRECTION']` blocks in `src/components/tooltip/_tooltip.scss`

File: packages/react/src/internal/FloatingMenu.js [hunk 2]
```diff
@@ -400,7 +400,7 @@ class FloatingMenu extends React.Component {
         ...styles,
         ...positioningStyle,
         position: 'absolute',
-        margin: 0,
+        // margin-bottom: 0,
```
Comment (tw15egan) [inline packages/react/src/internal/FloatingMenu.js]: Can we remove this comment?

**Review by laurenmrice — CHANGES_REQUESTED** [review]
- The left and top directions with start, center and end alignments still need `8px (spacing-03)` between the icon and the caret tip. The others look good !

File: packages/components/src/components/tooltip/_tooltip.scss [hunk 3]
```diff
@@ -514,6 +514,8 @@
     }
 
     &[data-floating-menu-direction='left'] {
+      margin-left: -$spacing-03;
```
Comment (tw15egan) [inline packages/components/src/components/tooltip/_tooltip.scss]: Looks like this is an issue with negative numbers and css custom properties. We can do either:

`margin-left: calc(#{$spacing-03} * -1)`

or

`margin-left: rem(-8px)`

File: packages/components/src/components/tooltip/_tooltip.scss [hunk 4]
```diff
@@ -524,6 +526,9 @@
     }
 
     &[data-floating-menu-direction='top'] {
+      margin-top: -$spacing-03;
```
Comment (tw15egan) [inline packages/components/src/components/tooltip/_tooltip.scss]: Same as above

**Comment by andreancardona** [comment]
@tw15egan @laurenmrice updated :)

**Review by laurenmrice — CHANGES_REQUESTED** [review]
- The top direction for start,center and end alignment needs to fixed still. The rest of the directions and alignments look great!

File: packages/components/src/components/tooltip/_tooltip.scss [hunk 5]
```diff
@@ -524,6 +526,8 @@
     }
 
     &[data-floating-menu-direction='top'] {
+      margin-left: calc(#{$spacing-03} * -1);
```
Comment (tw15egan) [inline packages/components/src/components/tooltip/_tooltip.scss]:
```suggestion
      margin-top: calc(#{$spacing-03} * -1);
```

**Review by laurenmrice — APPROVED** [review]
Looks good to me! 🎉

**Review by tw15egan — APPROVED** [review]
LGTM 👍 ✅

**Review by tay1orjones — APPROVED** [review]

**Pull request merged**